tsdb/index.go

```diff
@@ -1330,7 +1330,8 @@ func (is IndexSet) MeasurementNamesByExpr(auth query.FineAuthorizer, expr influx
 		if err != nil {
 			return nil, err
 		}
-		return slices.CopyChunkedByteSlices(names, 1000), nil
+		// Requires that names is a copy of the names from the mmapped file
+		return names, nil
```

Comment:
This seems quite different than before because we now allocate more memory for this part:

```
		case influxql.OR, influxql.AND:
			lhs, err := is.measurementNamesByPredicate(auth, e.LHS)
			if err != nil {
				return nil, err
			}
			rhs, err := is.measurementNamesByPredicate(auth, e.RHS)
			if err != nil {
				return nil, err
			}
```

Also what about the Expr versions - e.g. `measurementNamesByTagFilter` ?